generative/maisi/scripts/transforms.py

```diff
@@ -0,0 +1,292 @@
+    """
+    Define fixed intensity transform based on the modality.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        image_keys (List[str], optional): List of image keys. Defaults to ["image"].
+
+    Returns:
+        List: A list of intensity transforms.
+    """
+    intensity_transforms = {
+        "mri": [
+            ScaleIntensityRangePercentilesd(keys=image_keys, lower=0.0, upper=99.5, b_min=0.0, b_max=1, clip=False)
+        ],
+        "ct": [ScaleIntensityRanged(keys=image_keys, a_min=-1000, a_max=1000, b_min=0.0, b_max=1.0, clip=True)],
+    }
+
+    if modality not in intensity_transforms:
+        warnings.warn(f"No intensity transform is applied. Modality has to be in ['ct','mri']. Got {modality}.")
+        return []
+
+    return intensity_transforms[modality]
+
+
+def define_random_intensity_transform(modality: str, image_keys: List[str] = ["image"]) -> List:
+    """
+    Define random intensity transform based on the modality.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        image_keys (List[str], optional): List of image keys. Defaults to ["image"].
+
+    Returns:
+        List: A list of random intensity transforms.
+    """
+    if modality == "ct":
+        return []  # CT HU intensity is stable across different datasets
+    elif modality == "mri":
+        return [
+            RandBiasFieldd(keys=image_keys, prob=0.3, coeff_range=(0.0, 0.3)),
+            RandGibbsNoised(keys=image_keys, prob=0.3, alpha=(0.5, 1.0)),
+            RandAdjustContrastd(keys=image_keys, prob=0.3, gamma=(0.5, 2.0)),
+            RandHistogramShiftd(keys=image_keys, prob=0.05, num_control_points=10),
+        ]
+    else:
+        warnings.warn(f"No random intensity transform is applied. Modality has to be in ['ct','mri']. Got {modality}.")
+        return []
+
+
+def define_vae_transform(
+    modality: str,
+    data_aug: bool,
+    k: int = 4,
+    patch_size: List[int] = [128, 128, 128],
+    val_patch_size: Optional[List[int]] = None,
+    compute_dtype: torch.dtype = torch.float32,
+    spacing_type: str = "original",
+    spacing: Optional[List[float]] = None,
+    image_keys: List[str] = ["image"],
+    label_keys: List[str] = [],
+    additional_keys: List[str] = [],
+    select_channel: int = 0,
+) -> tuple:
+    """
+    Define the transform pipeline for training and validation.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        data_aug (bool): Whether to apply random data augmentation.
+        k (int, optional): Patches should be divisible by k. Defaults to 4.
+        patch_size (List[int], optional): Size of the patches. Defaults to [128, 128, 128].
+        val_patch_size (Optional[List[int]], optional): Size of validation patches. Defaults to None.
+        compute_dtype (torch.dtype, optional): Final data type. Defaults to torch.float32.
```

Comment:
I am confused about what the "final data" is when I first read this line. Does it involve multiple steps, or is it input/output? Maybe we can improve the docstring here.

Reply:
Output. changed docsring, thanks